static/src/javascripts/projects/commercial/modules/dfp/redplanet.js

```diff
@@ -1,38 +1,44 @@
 // @flow strict
 
 import config from 'lib/config';
-import { commercialFeatures } from "common/modules/commercial/commercial-features";
-import { onIabConsentNotification } from "@guardian/consent-management-platform";
-import { isInAuRegion } from "commercial/modules/header-bidding/utils";
-import { isInVariantSynchronous } from "common/modules/experiments/ab";
-import { commercialRedplanet } from "common/modules/experiments/tests/commercial-redplanet-aus";
+import { commercialFeatures } from 'common/modules/commercial/commercial-features';
+import { onIabConsentNotification } from '@guardian/consent-management-platform';
+import { isInAuRegion } from 'commercial/modules/header-bidding/utils';
+import { isInVariantSynchronous } from 'common/modules/experiments/ab';
+import { commercialRedplanet } from 'common/modules/experiments/tests/commercial-redplanet-aus';
+
+let initialised = false;
 
 const initialise = (): void => {
     // Initialise Launchpad Tracker
     window.launchpad('newTracker', 'launchpad', 'lpx.qantas.com', {
         discoverRootDomain: true,
-        appId: 'the-guardian'
+        appId: 'the-guardian',
     });
 
     // Track Page Views
     window.launchpad('trackUnstructEvent', {
-        'schema': 'iglu:com.qantas.launchpad/hierarchy/jsonschema/1-0-0',
-        'data': {
-            'u1': 'theguardian.com',
-            'u2':  config.get('page.section'),
-            'u3':  config.get('page.sectionName'),
-            'u4': config.get('page.contentType'),
-            'uid': config.get('ophan', {}).browserId,
-        }
+        schema: 'iglu:com.qantas.launchpad/hierarchy/jsonschema/1-0-0',
+        data: {
+            u1: 'theguardian.com',
+            u2: config.get('page.section'),
+            u3: config.get('page.sectionName'),
+            u4: config.get('page.contentType'),
+            uid: config.get('ophan', {}).browserId,
+        },
     });
 };
 
 const setupRedplanet: () => Promise<void> = () => {
     onIabConsentNotification(state => {
-        const consentState =
-            state[1] && state[2] && state[3] && state[4] && state[5];
+        // typeof state === 'boolean' means CCPA mode is on
+        const canRun =
+            typeof state === 'boolean'
+                ? !state
+                : state[1] && state[2] && state[3] && state[4] && state[5];
 
-        if (consentState) {
```

Comment:
Same here. Enforcement of CCPA for third party tags will be done in a separate card, as decisions on this have still to be taken.

Reply:
Plus Redplanet is already restricted to AU, and CCPA is in the US only.

Reply:
> Better to rely on the presence of window.__uspapi

this should probably come from the CMP somehow though. i feel like the less we sprinkle IAB objects through the code the better (less to worry about come tcfv3 etc).

should the callbacks be called with extra params to capture this logic? or is the logic specific to where it applies?

Reply:
Regarding relying on the state parameter: Agree. This is a half-way measure. The end goal is to release a new major version of the CMP lib where the API is slightly different and will pass both TCF and CCPA states separately (but only one will be available at any time). The reason why we chose to do that change later instead of now is that making that change would require us to alter our TCFv1 code as well, whereas this solution doesn't. Meaning this options leads to less dev time right now, putting us in a better position to meet our deadline. 
Can add a 'stateType' to the callback however, so it clearer, if people think it's worth it (considering that the current implementation is only going to be around for a few weeks)

Regarding Redplanet: understood. Will change to be more clear/accurate.

Regarding enforcement: I was under the impression that any 3rd party scripts that lead to a direct or indirect monetary gain should be blocked when CCPA's 'do not sell' flag is on and that might change in the future but for that that's how we are treating it. Is that not the case?

Reply:
Spoke 'in person' With Jerome and agree to change Redplanet's code and leave the other two points for now until we have further clarification.